### PR TITLE
Adding a run playbook to execute NDT E2E tests

### PR DESCRIPTION
**Pull request by mtlynch**
Adds a playbook to run NDT tests on each type of node (Windows, OS X, Linux).
Makes a minor permissions fix in the Linux prepare script so that non-root
user can access the Chrome Selenium driver.

---

This change is [Reviewable](https://reviewable.io/reviews/m-lab/ndt-e2e-ansible/7)

### CODE REVIEW

**Comment by mtlynch** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

_[run.yml, line 76 [r1]](https://reviewable.io:443/reviews/m-lab/ndt-e2e-ansible/7#-KH_mO5TeVTUbMFn2Kgv:-KH_mO5TeVTUbMFn2Kgw:-1867103390) ([raw file](https://github.com/m-lab/ndt-e2e-ansible/blob/4b84ae166fe84506ce3efd2f73bb34a9a91cddbc/run.yml#L76)):_

> ``` YAML
> #    variables.
> # Until Ansible has support for shell/command modules on Windows, this needs
> # to be a separate play.
> ```

Note on this: We could reduce redundancy a little bit if we generated the commands in advance with `set_fact`, then each play would just be something like `raw: "{{ html5_client_command }}"`, but this feels like too much indirection.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/ndt-e2e-ansible/7)_

**Comment by nkinkade** [comment]
Review status: 0 of 3 files reviewed at latest revision, 4 unresolved discussions.

---

_[run.yml, line 10 [r1]](https://reviewable.io:443/reviews/m-lab/ndt-e2e-ansible/7#-KHaIbb5H7QpuIWNSiIs:-KHaIbb5H7QpuIWNSiIt:-896329264) ([raw file](https://github.com/m-lab/ndt-e2e-ansible/blob/4b84ae166fe84506ce3efd2f73bb34a9a91cddbc/run.yml#L10)):_

> ``` YAML
>     - name: create result archive filename
>       set_fact:
>         archive_file: ndt-results-{{ ansible_hostname | lower }}-{{ ansible_date_time.iso8601 | regex_replace(":", "") }}.zip
> ```

Could this line be made a bit less ponderous by  using variables?

```
tags: facts
vars:
  hostname: "{{ ansible_hostname | lower }}"
  datetime: "{{ ansible_date_time.iso8601 | regex_replace(":", "") }}"
tasks:
  - name: create result archive filename
    set_fact:
      archive_file: ndt-results-{{ hostname }}-{{ datetime }}.zip
```

---

_[run.yml, line 102 [r1]](https://reviewable.io:443/reviews/m-lab/ndt-e2e-ansible/7#-KHaMeiSXf6sYDtkcBXh:-KHaMeiTppTZjvlpO9-6:1374708952) ([raw file](https://github.com/m-lab/ndt-e2e-ansible/blob/4b84ae166fe84506ce3efd2f73bb34a9a91cddbc/run.yml#L102)):_

> ``` YAML
>         --output={{ raw_results_dir }}
>         --iterations={{ iterations }}
>       with_items: "{{ supported_browsers }}"
> ```

Since we don't actually use `{{ item }}` any where in this task, would this cause the exact same task to be run len(supported_browsers) times?

---

_[run.yml, line 155 [r1]](https://reviewable.io:443/reviews/m-lab/ndt-e2e-ansible/7#-KHaOGS2RQiQ9BiDzZkr:-KHaOGS2RQiQ9BiDzZks:-1825222511) ([raw file](https://github.com/m-lab/ndt-e2e-ansible/blob/4b84ae166fe84506ce3efd2f73bb34a9a91cddbc/run.yml#L155)):_

> ``` YAML
> 
>     - name: move raw results to archive folder
>       shell: mv {{ raw_results_dir }}/* {{ archived_results_dir }}
> ```

Is it your intention to copy not only the ZIP file, but also all the unzipped files too? I believe the zip command above will zip all the files from `find` into the archive but will leave the originals in place... so this `mv` command will copy the zip archive as well as all the individual files, right?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/ndt-e2e-ansible/7)_

**Comment by mtlynch** [comment]
Review status: 0 of 3 files reviewed at latest revision, 4 unresolved discussions.

---

_[run.yml, line 10 [r1]](https://reviewable.io:443/reviews/m-lab/ndt-e2e-ansible/7#-KHaIbb5H7QpuIWNSiIs:-KHaW9lAM_396QpSO47r:143383958) ([raw file](https://github.com/m-lab/ndt-e2e-ansible/blob/4b84ae166fe84506ce3efd2f73bb34a9a91cddbc/run.yml#L10)):_
<details><summary>

<i>Previously, nkinkade wrote…</i></summary>



> Could this line be made a bit less ponderous by  using variables?
> 
> ```
> tags: facts
> vars:
>   hostname: "{{ ansible_hostname | lower }}"
>   datetime: "{{ ansible_date_time.iso8601 | regex_replace(":", "") }}"
> tasks:
>   - name: create result archive filename
>     set_fact:
>       archive_file: ndt-results-{{ hostname }}-{{ datetime }}.zip
> ```
> 
> </details>
> Good idea, fixed!

---

_[run.yml, line 102 [r1]](https://reviewable.io:443/reviews/m-lab/ndt-e2e-ansible/7#-KHaMeiSXf6sYDtkcBXh:-KHaWECMKu9W_LNSIUFi:478262235) ([raw file](https://github.com/m-lab/ndt-e2e-ansible/blob/4b84ae166fe84506ce3efd2f73bb34a9a91cddbc/run.yml#L102)):_
<details><summary>

<i>Previously, nkinkade wrote…</i></summary>



> Since we don't actually use `{{ item }}` any where in this task, would this cause the exact same task to be run len(supported_browsers) times?</details>
> Oh good catch. That was actually a mistake and `{{ item }}` should be there. I think I got lazy when trying to debug something and just hardcoded the command instead of overriding with `--extra-vars`. Fixed.

---

_[run.yml, line 155 [r1]](https://reviewable.io:443/reviews/m-lab/ndt-e2e-ansible/7#-KHaOGS2RQiQ9BiDzZkr:-KHaWxhEwetEzyLZinfM:-137267119) ([raw file](https://github.com/m-lab/ndt-e2e-ansible/blob/4b84ae166fe84506ce3efd2f73bb34a9a91cddbc/run.yml#L155)):_
<details><summary>

<i>Previously, nkinkade wrote…</i></summary>



> Is it your intention to copy not only the ZIP file, but also all the unzipped files too? I believe the zip command above will zip all the files from `find` into the archive but will leave the originals in place... so this `mv` command will copy the zip archive as well as all the individual files, right?</details>
> I think this might not be coming across as clearly as I was hoping. It might help if we walk through an example. It should look like this:

``` bash
cd /opt/ndt-results/raw; find . | \
  zip /opt/ndt-results/zips/ndt-results-mlab-linux-mini-2016-05-12T155624Z.zip -@
mv /opt/ndt-results/raw/* /opt/ndt-results/archived/
```

In other words, once we zip the raw files, we move them to the archived results folder. This is essentially just a backup in case something gets screwed up with the package and we want to see all the old results in raw form.

I've updated the task names to try to clarify this a bit.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/ndt-e2e-ansible/7)_

**Comment by nkinkade** [comment]
:lgtm:

<details><summary>

<i>Previously, mtlynch (Michael Lynch) wrote…</i></summary>



> #### Adding a run playbook to execute NDT E2E tests
> 
> Adds a playbook to run NDT tests on each type of node (Windows, OS X, Linux).
> 
> Makes a minor permissions fix in the Linux prepare script so that non-root
> 
> user can access the Chrome Selenium driver.
> 
> </details>

---

Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_[run.yml, line 155 [r1]](https://reviewable.io:443/reviews/m-lab/ndt-e2e-ansible/7#-KHaOGS2RQiQ9BiDzZkr:-KHam7tMAHOtMLySJO0q:-470199613) ([raw file](https://github.com/m-lab/ndt-e2e-ansible/blob/4b84ae166fe84506ce3efd2f73bb34a9a91cddbc/run.yml#L155)):_
<details><summary>

<i>Previously, mtlynch (Michael Lynch) wrote…</i></summary>



> I think this might not be coming across as clearly as I was hoping. It might help if we walk through an example. It should look like this:
> 
> ``` bash
> cd /opt/ndt-results/raw; find . | \
>   zip /opt/ndt-results/zips/ndt-results-mlab-linux-mini-2016-05-12T155624Z.zip -@
> mv /opt/ndt-results/raw/* /opt/ndt-results/archived/
> ```
> 
> In other words, once we zip the raw files, we move them to the archived results folder. This is essentially just a backup in case something gets screwed up with the package and we want to see all the old results in raw form.
> 
> I've updated the task names to try to clarify this a bit.</details>
> Ah, got it. I think my confusion was with `archive_path`, which is to a totally different filesystem location.  That is, the ZIP archive isn't created in the `raw_results_dir`.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/ndt-e2e-ansible/7#-:-KHamiVJGn6E1nLQ2W7Q:1417088721)_
